File: Level-1/64.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        int[] arr = new int[number + 1];
        for(int i = 1; i <= number; i++) {
            for(int j = 1; j <= Math.sqrt(i); j++) {
                if(i % j == 0) {
                    if(i / j == j) {
                        arr[i]++;
                    }else {
                        arr[i] += 2;
                    }
                }
            }
        }
        
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] > limit) {
                answer += power;
            }else {
                answer += arr[i];
            }
        }
        
        return answer;
    }
}

File: Level-1/64.js
function solution(number, limit, power) {
    const attack = Array.from(new Array(number + 1), () => 0);
    
    for(let i = 1; i <= number; i++) {
        for(let j = 1; j <= Math.sqrt(i); j++) {
            if(i % j === 0) {
                (j === i / j) ? attack[i] += 1 : attack[i] += 2;
            }
        }
        
        if(attack[i] > limit) attack[i] = power;
    }
    
    return attack.reduce((acc, cur) => acc + cur, 0);
}
